src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/util/ProfessionalReferenceDataClient.java

```diff
@@ -67,8 +68,15 @@ public ProfessionalReferenceDataClient(int port, String issuer, Long tokenExpira
         return getRequest("/refdata/internal/v1/organisations" + "/pbas?email={email}", role, email);
     }
 
-    public Map<String, Object> findPaymentAccountsByEmailForExternalUsers(String email, String role) {
-        return getRequest("/refdata/external/v1/organisations" + "/pbas?email={email}", role, email);
+    public Map<String, Object> findPaymentAccountsByEmailFromHeader(String email, String role) {
+        return getRequestToGetEmailFromHeader("/refdata/internal/v1/organisations" + "/pbas?email={email}",
+                role, "", email);
+    }
+
+    public Map<String, Object> findPaymentAccountsByEmailFromHeaderForExternalUsers(String email, String role,
+                                                                                    String userId) {
```

Comment:
should not you have two methods here. -
1. one which takes parameter from header (with no email id as query param)
2. another method which takes the email id parameter from query and having a check of param from header
and there should be subsequent test cases using each of the above?

Right now it tests only the scenario when both conditions - from header and query param are checked  and the final state where  only the header param would exist is not tested at all

Reply:
as discussed closing the conversation